priam/src/main/java/com/netflix/priam/google/GoogleEncryptedFileSystem.java

```diff
@@ -274,59 +208,27 @@ public void upload(AbstractBackupPath path, InputStream in)
     @Override
     public void cleanup() {
         // TODO Auto-generated method stub
-
-    }
-
-    @Override
-    public int getActivecount() {
-        // TODO Auto-generated method stub
-        return 0;
-
     }
 
     @Override
     public void shutdown() {
         // TODO Auto-generated method stub
-
     }
 
     @Override
-    public int downloadCount() {
-        return this.downloadCount.get();
-    }
-
-    @Override
-    public int uploadCount() {
-        // TODO Auto-generated method stub
-        return 0;
-    }
-
-    @Override
-    public long bytesUploaded() {
-        // TODO Auto-generated method stub
-        return 0;
-    }
-
-    @Override
-    public long getBytesUploaded() {
-        return 0;
+    protected long uploadFileImpl(Path localPath, Path remotePath) throws BackupRestoreException {
+        throw new UnsupportedOperationException();

```

Comment:
wait this is unsupported?

Reply:
So far we have only supported download from Google File System (GFS). Hopefully, in future, we can support GFS in the same way as S3 (all operations) by hiding the complexity of encryption and compression.